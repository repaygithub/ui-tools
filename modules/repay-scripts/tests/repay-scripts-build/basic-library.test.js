const TestSetup = require('../test-setup')

beforeEach(() => jest.setTimeout(1000 * 60 * 5))

describe('@repay/repay-scripts', () => {
  it('should build a basic library', async () => {
    const test = new TestSetup('basic-library')
    await test.setup()
    const pkg = JSON.parse(await test.readFile('package.json'))
    await test.exec(`yarn repay-scripts build --lib index.js`)
    expect(await test.readFile(pkg.main)).toMatchSnapshot()
    expect(await test.readFile(pkg.module)).toMatchSnapshot()
  })
})
