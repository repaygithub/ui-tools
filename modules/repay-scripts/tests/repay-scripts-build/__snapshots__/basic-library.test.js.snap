// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@repay/repay-scripts should build a basic library 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));

function App(props) {
  return React.createElement(\\"div\\", {
    className: \\"AppWrapper\\"
  }, props.children);
}

exports.App = App;
//# sourceMappingURL=build.js.map
"
`;

exports[`@repay/repay-scripts should build a basic library 2`] = `
"import React from 'react';

function App(props) {
  return React.createElement(\\"div\\", {
    className: \\"AppWrapper\\"
  }, props.children);
}

export { App };
//# sourceMappingURL=build.esm.js.map
"
`;
